### ISSUE TITLE
[plan] Agent-opened pull requests start no CI: set AGENT_GITHUB_TOKEN and wire agent-02-execute.yml to it

### ISSUE THREAD

**Issue by stardustsuperwizard** [stardustsuperwizard/mikeys_game_bones-rules-moba#394]
## Goal

Every pull request opened by an agent workflow should start the same
`pull_request` gates a human-opened one does — Godot validation, gdscript lint,
issue linking, dependency checks — so that "CI is green" means the gates ran and
passed, rather than that they never ran.

## Current State

GitHub does not trigger workflows from events attributed to `GITHUB_TOKEN`. It is
a deliberate loop-prevention rule, not a misconfiguration. Two workflows open
pull requests as that identity:

- `agent-06-claude.yml` reads `GH_TOKEN: ${{ secrets.AGENT_GITHUB_TOKEN || secrets.GITHUB_TOKEN }}`
  (L135 for checkout, L567 for the session). The escape hatch exists, but
  `AGENT_GITHUB_TOKEN` is unset, so it falls through to `GITHUB_TOKEN`.
- `agent-02-execute.yml` sets `GH_TOKEN: ${{ github.token }}` / `GITHUB_TOKEN: ${{ github.token }}`
  (L134-135) and runs `gh pr create`. It has no `AGENT_GITHUB_TOKEN` fallback at
  all, so setting the secret alone does not fix this one.

Both are already documented as known: the identity comment above the session step
in `agent-06-claude.yml` (L526-546) states the condition, names the fix, and notes
that `agent-02-execute.yml` shares it.

The result is that a `claude:execute` or `agent:execute` PR carries an empty check
list. Nothing is red, so it reads as passing at a glance, while `godot-ci-validation.yml`
and `gdscript-lint.yml` have not run against the diff at all.

## Desired State

```
agent workflow
→ opens PR as AGENT_GITHUB_TOKEN
→ GitHub fires pull_request
→ godot-ci-validation.yml + gdscript-lint.yml + issue-linking.yml run
→ checks appear on the PR
→ green means the gates passed
```

## Requirements

- Create an `AGENT_GITHUB_TOKEN` repository secret: a PAT or GitHub App
  installation token carrying `contents: write` and `pull-requests: write`.
- Give `agent-02-execute.yml` the same `secrets.AGENT_GITHUB_TOKEN || github.token`
  fallback `agent-06-claude.yml` already has, at both its checkout and its
  `gh pr create` call site, so the two workflows behave identically.
- Keep the fallback rather than requiring the secret. Unset, behaviour must stay
  exactly as it is today — a fork or a clean clone should not start failing runs
  because a secret it cannot have is missing.
- Update the setup notes in `agent-06-claude.yml` (L48-54) if the secret stops
  being optional in practice for the execute path.

### Success

1. A `claude:execute` or `agent:execute` run opens a PR as `AGENT_GITHUB_TOKEN`.
2. GitHub fires `pull_request`, and the path-matching gates start.
3. The PR shows real check runs; a red one blocks it the way it would on a
   human-opened PR.

### Failure

1. `AGENT_GITHUB_TOKEN` is unset: the workflows fall back to `GITHUB_TOKEN` and
   behave as they do today, with no new failure introduced.
2. The token is set but under-scoped: `git push` or `gh pr create` fails loudly in
   the run log rather than opening a PR that silently starts nothing.
3. Failure reason is visible to the operator/agent.

## Constraints

- Prefer deterministic automation over agent reasoning.
- Do not introduce persistent infrastructure unless required.
- Do not store long-lived credentials in the repository.
- Keep development setup and independent validation separate where applicable.

Also specific to this change:

- `GITHUB_TOKEN` cannot write `.github/workflows/**` or `.github/actions/**`
  regardless of the `permissions:` key — GitHub reserves that for credentials
  carrying the `workflow` scope. `agent-02-execute.yml` already handles this
  explicitly (L280, L375). Scoping `AGENT_GITHUB_TOKEN` decides whether that
  restriction still applies, so it is a deliberate choice, not an oversight to
  fix in passing.

## Acceptance Criteria

- [ ] `AGENT_GITHUB_TOKEN` exists as a repository secret with `contents: write`
      and `pull-requests: write`.
- [ ] `agent-02-execute.yml` uses `secrets.AGENT_GITHUB_TOKEN || github.token` at
      its checkout and its `gh pr create` call site.
- [ ] An agent-opened PR shows check runs from `godot-ci-validation.yml` and/or
      `gdscript-lint.yml` when its diff matches their path filters.
- [ ] With the secret unset, both workflows still run and still open PRs — only
      the missing checks come back.
- [ ] Failure path has been verified.
- [ ] Documentation updated where necessary.

## Out of Scope

- Changing which gates run, or their path filters.
- Any change to how `agent-04-review.yml` or `agent-05-fix.yml` are triggered.
- Reworking the review/fix contracts in `.claude/` or `.github/agents/`.

## Dependencies

| Relationship | Issue | Why |
| --- | --- | --- |
| None | — | — |

---

Found while implementing #393, which teaches `/execute-task` to wait for CI on the
PR it opens. That step has to distinguish "no checks because the diff is
path-filtered out" (green) from "no checks because the author cannot start
workflows" (not green), and the second case only exists because of this gap. #393
reports the condition correctly; this Issue removes it.
